tests/unit/components/Collections/Table/Table.spec.js

```diff
@@ -1,9 +1,240 @@
 import Table from "@/components/Collections/Table/Table";
+import { setMatchMedia } from "@/services/Testing/Utilities";
+import store from "@/store";
+
+var dummyTableData = [
+  {
+    name: { type: "component", value: "123", status: "draft" },
+    number_of_viewers: "0",
+  },
+  {
+    name: { type: "component", value: "234", status: "published" },
+    number_of_viewers: "10",
+  },
+];
+
+const tableColumns = ["name", "number_of_viewers"];
+const totalNumberOfPlios = dummyTableData.length;
 
 describe("Table.vue", () => {
+  beforeEach(async () => {
+    await store.dispatch("sync/stopLoading");
+    await setMatchMedia(false);
+  });
+
   it("should render with default values", () => {
     const wrapper = mount(Table);
-    expect(wrapper).toBeTruthy();
+    expect(wrapper.vm.totalItemsInTable).toBe(0);
+    expect(wrapper.vm.isTableEmpty).toBe(true);
+  });
+
+  it("renders the right number of rows ", async () => {
+    const wrapper = mount(Table, {
+      props: {
+        data: dummyTableData,
+        columns: tableColumns,
+        numTotal: totalNumberOfPlios,
+      },
+    });
+    expect(wrapper.findAll("th").length).toBe(dummyTableData.length);
+    expect(wrapper.vm.totalItemsInTable).toBe(dummyTableData.length);
+    expect(wrapper.vm.isTableEmpty).toBe(false);
+
+    // number of rows * number of columns (the button is rendered for each columns
+    // but is hidden on all columns except the last column)
+    expect(wrapper.findAll('[data-test="analyzeButton"]').length).toBe(4);
+  });
+
+  it("does not render analyze on phone ", async () => {
+    // set `matches` as `True` for testing on touch screen devices
+    setMatchMedia(true);
+
+    const wrapper = mount(Table, {
+      props: {
+        data: dummyTableData,
+        columns: tableColumns,
+        numTotal: totalNumberOfPlios,
+      },
+    });
+    expect(wrapper.findAll('[data-test="analyzeButton"]').length).toBe(0);
+  });
+
+  it("does not render analyze on pending", async () => {
+    const wrapper = mount(Table, {
+      props: {
+        data: dummyTableData,
+        columns: tableColumns,
+        numTotal: totalNumberOfPlios,
+      },
+      data() {
+        return {
+          selectedRowIndex: 0,
+        };
+      },
+    });
+
+    store.dispatch("sync/startLoading");
+    expect(wrapper.vm.tableCellOverlayClass(0, 1)).toEqual({ hidden: true });
+  });
+
+  it("analyze button should be disabled for draft plio ", async () => {
+    const wrapper = mount(Table, {
+      props: {
+        data: dummyTableData,
+        columns: tableColumns,
+        numTotal: totalNumberOfPlios,
+      },
+    });
+    // the button is rendered for each columns but is hidden on all columns
+    // except the last column
+    expect(
+      wrapper.findAll('[data-test="analyzeButton"]')[1].element.disabled
+    ).toBe(true);
+    expect(
+      wrapper.findAll('[data-test="analyzeButton"]')[3].element.disabled
+    ).toBe(false);
```

Comment:
please explain

Reply:
comment saying draft - disabled, published - not disabled

Reply:
num rows (2) x num cols (2)